packages/fuel-indexer/src/executor.rs

```diff
@@ -131,6 +132,15 @@ pub fn run_executor<T: 'static + Executor + Send + Sync>(
 
             let mut block_info = Vec::new();
             for block in results.into_iter() {
+                if let Some(end_block) = end_block {
+                    if block.header.height.0 > end_block {
+                        info!("Stopping indexer at the specified end_block: {end_block}");
+                        break;
+                    }
+                } else {
+                    warn!("No end block is set. Indexer will run forever!");
```

Comment:
This warning will print for every block? I don't think we want that. We can remove it (we can add the warning somewhere else, because I do think we should warn when indexers run forever).